### PR TITLE
Started to port a few things from Spectre to SpectreKit

### PR DESCRIPTION
**Pull request by migueldeicaza**
Decided to do a few things this Sunday morning, I am trying to stick to your design and style.

### CODE REVIEW

File: Sources/SpectreKit/Rendering/Segments.swift [hunk 1]
```diff
@@ -107,6 +107,42 @@ public enum Segment: Equatable {
...
+            // TODO: if builder.count == 0 should we return .empty?
+            return .text(content: builder, style: style)
+        }
+        switch self {
+        case .controlSequence(code: _):
+            fatalError("TODO")
+        case .empty:
+            return self
+        case .lineBreak:
+            fatalError("TODO: I think this should be empty")
```
Comment (patriksvensson) [inline Sources/SpectreKit/Rendering/Segments.swift]: I think we can return an empty segment here

File: Sources/SpectreKit/Rendering/Segments.swift [hunk 2]
```diff
@@ -107,6 +107,42 @@ public enum Segment: Equatable {
...
+                    break
+                }
+                builder.append(character)
+            }
+            // TODO: if builder.count == 0 should we return .empty?
+            return .text(content: builder, style: style)
+        }
+        switch self {
+        case .controlSequence(code: _):
+            fatalError("TODO")
```
Comment (patriksvensson) [inline Sources/SpectreKit/Rendering/Segments.swift]: I think we can return the segment as-is here

**Review by patriksvensson — CHANGES_REQUESTED** [review]
Looks good to me! The only two things that need to be fixed are the TODO parts in `Segment.truncate`.

Also, don't hesitate to do things in a more "Swiftesque" way if you think of something. I'm trying to learn, and if there is a more idiomatic way of doing things than I've done so far, I'm happy to change it. Also, SpectreKit does not have to be a 1:1 representation of Spectre.Console, so feel free to be creative 🙂

**Comment by patriksvensson** [comment]
@migueldeicaza FYI, I've merged proper cell calculation to main, so you might want to rebase.

**Comment by migueldeicaza** [comment]
Thank you for your review and guidance Patrik!

**Comment by patriksvensson** [comment]
Merged! Thank you for your contribution. Much appreciated! 👍

**Pull request merged**